R/plotLoadings.R

```diff
@@ -0,0 +1,386 @@
+#' tse <- addLDA(tse)
+#' plotLoadings(tse, dimred = "LDA", ncomponents = 2)
+NULL
+
+#' @rdname plotLoadings
+setGeneric("plotLoadings", signature = c("x"),
+    function(x, ...) 
+        standardGeneric("plotLoadings"))
+
+
+#' @rdname plotLoadings
+#' @export 
+setMethod("plotLoadings", signature = c(x = "TreeSummarizedExperiment"),
+    function(x,
+            dimred,
+            layout = "barplot",
+            n = 10,
+            ncomponents = 5,
+            tree.name = "phylo",
+            rank = NULL,
+            add.tree = FALSE,
+            ...) {
+      
+        # Making sure there is no error in parameters given by the user
+        .check_parameters(x,
+                        dimred = dimred,
+                        layout = layout,
+                        n = n,
+                        ncomponents = ncomponents,
+                        tree.name = tree.name,
+                        rank = rank,
+                        add.tree = add.tree,
+                        ...)
+        loading_names <- c("rotation", "loadings")
+        attr_names <- names(attributes(reducedDim(x, dimred)))
+        attr_name <- attr_names[ attr_names %in% loading_names ]
+        if( length(attr_name) != 1 ) {
+            stop("Loadings cannot be found..", call. = FALSE)
+        }
+        loadings_matrix <- attr(reducedDim(x, dimred), attr_name)
+
+        
+        # Checking if there are enough components in the matrix
+        .check_components(loadings_matrix, ncomponents)
+        
+        if (add.tree && layout == "heatmap") {
+            # Plot tree with feature loadings
+            p <- .loadings_tree_plotter(x, loadings_matrix, ncomponents, tree.name, rank)
+        } else if (layout == "heatmap") {
+            # Plot features with heatmap layout
+            p <- ComplexHeatmap::Heatmap(loadings_matrix, heatmap_legend_param = list(title = "Value")) 
+        } else {
+            # Ordering loadings and adding factor to keep the order
+            df <- .get_loadings_plot_data(loadings_matrix, n, ncomponents)
+            # Plot features with barplot layout
+            p <- .barplot_feature_loadings(df)
+        }
+    return(p)
+    }
+)
+#' @rdname plotLoadings
+#' @export 
+setMethod("plotLoadings", signature = c(x = "matrix"),
+    function(x,
+            layout = "barplot",
+            n = 10,
+            ncomponents = 5,
+            ...) {
+      
+        # Making sure there is no error in parameters given by the user
+        .check_parameters(x,
+                        layout = layout,
+                        n = n,
+                        ncomponents = ncomponents,
+                        ...)
+                      
+        # Checking if there are enough components in the matrix
+        .check_components(x, ncomponents)
+        
+        # Plot features with heatmap layout
+        if (layout == "heatmap") {
+            p <- ComplexHeatmap::Heatmap(x, heatmap_legend_param = list(title = "Value"))
```

Comment:
This looks good, but the problem is that all other methods output ggplot objects. --> create a function that gets data from

.get_loadings_plot_data(x, n, ncomponents)

as input and returns heatmap created with ggplot2